assets/src/amp-block-editor.js

```diff
@@ -95,6 +101,41 @@ function ComposedAMPToggle() {
 	] )( AMPToggle );
 }
 
+/**
+ * Whether the image has the minimum width for a featured image.
+ *
+ * This should have a width of at least 1200 pixels
+ * to satisfy the requirement of Google Search for Schema.org metadata.
+ *
+ * @param {Object} media A media object with width and height values.
+ * @return {boolean} Whether the media has the minimum dimensions.
+ */
+const hasMinimumFeaturedImageWidth = ( media ) => {
+	return ( media.width && media.width >= 1200 );
```

Comment:
If it's alright, I used an entirely different function for normal posts than the AMP Story featured image. The logic was quite different.